### PR TITLE
Components read right

### PR DESCRIPTION
**Pull request by btry**
This is a proposal to allow read only on components; see #1132

### CODE REVIEW

**Comment by btry** [comment]
@remicollet : can you review the migratioin code please ?

**Comment by remicollet** [comment]
So this mean either we need an update_91_911, or differ this change to 9.2

**Comment by btry** [comment]
Hi

I got misunderstanding about which file should contain a migration procedure. My apologies. 

I created a new update_xx_xx file and moved here the migration procedure.

**Comment by orthagh** [comment]
This need to be merge (and "deconflicted") after #1265 merged

**Comment by orthagh** [comment]
bump, #1285 merged

**Comment by orthagh** [comment]
@btry.
When you'll have time, could you update this pr (and move the update process to the correct file) ?

**Comment by btry** [comment]
@orthagh @remicollet : PR updated

Please review it and feedback.

File: inc/commondevice.class.php [hunk 1]
```diff
@@ -105,7 +99,7 @@ static function getItem_DeviceType() {
    static function getMenuContent() {
 
       $menu = array();
-      if (Session::haveRightsOr('device', array(CREATE, UPDATE, PURGE))) {
+      if (CommonDevice::canView()) {
```
Comment (orthagh) [inline inc/commondevice.class.php]: self::

File: install/update_911_912.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      }
+   }
+   if ($backup_tables) {
+      $migration->displayWarning("You can delete backup tables if you have no need of them.",
+            true);
+   }
+
+   // give READ right on components to profiles having UPDATE right
+   $query = "UPDATE `glpi_profilerights`
+             SET `rights` = `rights` | 1
```
Comment (orthagh) [inline install/update_911_912.php]: use constants
(READ, UPDATE)

File: install/update_911_912.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                   AND `name` = 'device'";
+   $DB->queryOrDie($query, "grant READ right on components to profiles having UPDATE right");
+
+   //put you migration script here
+
+   // ************ Keep it at the end **************
+   $migration->executeMigration();
+
+   return $updateresult;
+}
```
Comment (orthagh) [inline install/update_911_912.php]: missing CR

**Comment by orthagh** [comment]
as discussed together, this pr needs to move to master

**Comment by btry** [comment]
Due to his nature, this PR will not merge in 9.1.2, but in 9.2

**Pull request merged**